### PR TITLE
Images service

### PR DESCRIPTION
**Pull request by Gouga34**
POST images metadata
POST image (file)
GET images (search by : uri, rdf type, date, concerned elements)

### CODE REVIEW

File: phis2-ws/src/main/java/phis2ws/service/view/brapi/form/ResponseFormImageMetadata.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package phis2ws.service.view.brapi.form;
+
+import java.util.ArrayList;
+import phis2ws.service.view.brapi.Metadata;
+import phis2ws.service.view.brapi.results.ResultImageMetadata;
+import phis2ws.service.view.manager.ResultForm;
+import phis2ws.service.view.model.phis.ImageMetadata;
+
+/**
+ * Allows the formating of the result of the reqyest about Image Metadata
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/view/brapi/form/ResponseFormImageMetadata.java]: "reqyest"
Maybe change brapi package ? package phis2ws.service.view.brapi.form
?issue ?

File: phis2-ws/src/main/java/phis2ws/service/view/model/phis/ImageMetadata.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import java.util.ArrayList;
+
+/**
+ * Represents the image metadata view
+ * @author Morgane Vidal <morgane.vidal@inra.fr>
+ */
+public class ImageMetadata {
+    
+    private String uri;
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/view/model/phis/ImageMetadata.java]: comment

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/ImageMetadataDaoMongo.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+import phis2ws.service.view.model.phis.ShootingConfiguration;
+
+/**
+ * Represents the MongoDB Data Access Object for the images
+ * @author Morgane Vidal <morgane.vidal@inra.fr>
+ */
+public class ImageMetadataDaoMongo extends DAOMongo<ImageMetadata> {
+
+    final static Logger LOGGER = LoggerFactory.getLogger(ImageMetadataDaoMongo.class);
+    public String uri;
```
[the remark this answers is not in the document]
Comment (Gouga34) [reply phis2-ws/src/main/java/phis2ws/service/dao/mongo/ImageMetadataDaoMongo.java]: I agree with this idea. I open an issue

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/DatasetDAOMongo.java [hunk 4]
```diff
@@ -200,49 +226,72 @@ private POSTResultsReturn checkAndInsertPhenotypesList(ArrayList<PhenotypeDTO> p
...
-                LOGGER.trace("MongoDB insert : " + provenance.toJson());
-                
-                provenanceCollection.insertOne(provenance);
+            //transactions
+            //mongodb insertion
+            for (Dataset dataset : datasets) {
+                //1. provenance creation if needed
+                Object provenanceId = null;
+                if (dataset.getProvenance().getCreationDate() != null) {
+                    Document provenanceDocument = new Document();
```
[the remark this answers is not in the document]
Comment (Gouga34) [reply phis2-ws/src/main/java/phis2ws/service/dao/mongo/DatasetDAOMongo.java]: I have open an issue :  #24

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/ImageMetadataDaoMongo.java [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+    public String rdfType;
+    public String date;
+    public ArrayList<String> concernedItems = new ArrayList<>();
+    
+    private final MongoCollection<Document> imagesCollection = database.getCollection(PropertiesFileManager.getConfigFileProperty("mongodb_nosql_config", "images"));
```
[the remark this answers is not in the document]
Comment (Gouga34) [reply phis2-ws/src/main/java/phis2ws/service/dao/mongo/ImageMetadataDaoMongo.java]: it is the name of the configuration variable in the mongodb_nosql_config file, does it really needs a constant ?

File: phis2-ws/src/main/java/phis2ws/service/view/brapi/form/ResponseFormImageMetadata.java (hunk 1 above)
Comment (Gouga34) [reply phis2-ws/src/main/java/phis2ws/service/view/brapi/form/ResponseFormImageMetadata.java]: I opened an issue (#25)

**Pull request merged**